SwissKnife/avro/AvroTransformer.py

```diff
@@ -0,0 +1,271 @@
+import re
+from typing import Callable
+from typing import Dict
+
+# The type of each record file.
+Record = Dict[str, object]
+
+
+class NoDefault(object):
+    pass
+
+
+class AvroTransformer(object):
+    """
+    Avrotransformer provides some useful methods to
+    transform a data list (List[dict]) into another one
+    using an avro schema as reference to do transformations.
+
+    This avro schema will have some custom fields:
+        - transform: Indicates a transform function that will be applied
+          to input value to get the correct one. This transform can be a
+          type conversion, scaling, a threshold, etc.
+        - comment: Some info of the field.
+    """
+
+    def __init__(self, avro_schema: dict):
+        """Constructor of an AvroTransformer. Use an avro schema
+        as a reference to rename and trasform records.
+        :param avro_schema: The provided schema.
+        :type avro_schema: dict
+        """
+        self.original_schema = avro_schema
+        self.rename_dict = AvroTransformer._create_rename_dict(avro_schema)
+        self.defaults_dict = AvroTransformer._create_defaults_dict(avro_schema)
+        self.transform_dict = AvroTransformer._create_transform_dict(avro_schema)
+        self.cast_dict = AvroTransformer._create_cast_dict(avro_schema)
+
+    def get_original_schema(self) -> dict:
+        """Returns the original provided schema.
+        :return: An avro schema.
+        :rtype: dict
+        """
+        return self.original_schema
+
+    def apply_all_transforms(self, record: Record) -> Record:
+        """This function applies all the transformations of this object to a
+           input record.
+           The steps will be:
+            - get_renamed_record_an_remove_invalid_fields
+            - get_transformed_record
+            - get_record_with_defaults
+        :param record: The input record.
+        :type record: Record
+        :return: A record with all transformations applied.
+        :rtype: Record
+        """
+        renamed_record = self.get_renamed_record_and_remove_invalid_fields(record)
+        transformed_record = self.get_transformed_record(renamed_record)
+        record_with_defaults = self.get_record_with_defaults(transformed_record)
+        record_with_casted_values = self.get_record_with_casted_values(record_with_defaults)
```

Comment:
I'd just return in this line, instead of creating variable `record_with_casted_values`